Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -164,7 +164,7 @@ extension FileManager {
                 delegateResponse = delegate.fileManager?(self, shouldProceedAfterError: error, removingItemAtPath: path)
             }
             #else
-            delegateResponse = delegate.fileManager(self, shouldProceedAfterError: error, removingItemAtPath: path)
```

Comment:
In the FOUNDATION_FRAMEWORK branch above, we fall back to the `String`-based delegate functions after failing the `URL`-based delegate functions. Do we want to do the same here?

Reply:
In the `FOUNDATION_FRAMEWORK` branch, we fallback when the delegate does not implement the URL-based functions and we can detect this because it's an ObjC `@optional` protocol requirement. Since we don't have that same construct here, we can't differentiate between the URL-based function returning an explicit value or the default value. Instead, this patch updates the URL-based function's default implementation to call the String-based function, so if the client only implements the String-based function, calling the URL-based function here will call through to the String based one (which will return the default value if neither are implemented)

Reply:
Didn't realize these are `optional` in obj-c, and yeah having the default implementation in Swift for URL-based ones definitely helps. Thanks that makes sense